lib/phoenix/router.ex

```diff
@@ -676,7 +676,12 @@ defmodule Phoenix.Router do
   See `pipeline/2` for more information.
   """
   defmacro plug(plug, opts \\ []) do
-    plug = Macro.expand(plug, %{__CALLER__ | function: {:init, 1}})
+    plug =
+      if Phoenix.plug_init_mode() == :runtime do
```

Comment:
we have a couple other calls to this function https://github.com/phoenixframework/phoenix/blob/2e663e28934a81635e82c7e494fcb15d4e86e76a/lib/phoenix/router.ex#L534, https://github.com/phoenixframework/phoenix/blob/2e663e28934a81635e82c7e494fcb15d4e86e76a/lib/phoenix/router.ex#L647, the ets lookup is of course plenty fast but I wonder if conceptually it would make sense to set it just once in a module attribute and just refer to it.

Reply:
actually scratch that, it might be too restrictive if we want to test things and in some weird meta programming situations :)